lib/text_order/rspec.rb

```diff
@@ -1,11 +1,21 @@
 require 'text_order/matcher'
+require 'rspec/matchers/built_in/compound'
 
 module TextOrder
   module RSpec
     module Matchers
       def include_text(expected)
         Matcher.new(expected)
       end
+
+      def include_in_order(*texts)
+        texts
+          .each_cons(2)
+          .map { |(left, right)| Matcher.new(left).before(right) }
+          .reduce { |m1, m2|
```

Comment:
This reduce method here is pretty interesting. `m1` is the accumulator, and the block it returns the new instance of `And` making `m1` that instance in the next iteration. Is this the way rspec chains together matchers internally? Param 1 is the previous matcher and param 2 is the next one?  And they both need to have some kind of `Matcher` interface I guess.

Reply:
Ah yep, looks like it!  https://github.com/rspec/rspec-expectations/blob/master/lib/rspec/matchers/composable.rb#L23

Very cool!  Miss working with you @iamvery

Reply:
That's right! I thought it came together pretty interestingly. A composition is matchers ✨ . I miss you! 😊